### DISCUSSION TITLE
Calulation of the rate?

### DISCUSSION THREAD

**Discussion by DonKracho** [jomjol/AI-on-the-edge-device#1212] [Q&A]
I just installed this great project to my water meter. It counts m³ and has 5 digits for [m³] and 4 analog pointers for fractions. The last pointer has the unit 0,0001 m³ = 0.1l. I configured measurements to be done every 5 minutes, what results in 12 measurements in a hour.

When the meter value increases by 0.0001 between two measurements I expect the rate to be 0.0001 * 60 /  5 = 0.0012 m³/h (1.2 l/h) but it is shown as 0.00002 m³/h.

How is the rate calculated? Did I miss to configure something?

**Comment by jomjol** [comment]
You can choose between absolut or change rate. Up to now it seems to be okay.
Can you post parts of a log file, showing the wrong calculated values?

**Comment by DonKracho** [comment]
Thanks for the quick response. Currently I'm facing issues with the WiFi connection and random reboots. The ESP32-CAM tends to reboot when accessing it via the web interface and a new image captured shall be shown. The canvas remains white an it reboots. Meanwhile I increased the update rate to 3min, to get an easier to calulate rate factor of 20 instead of 12 for 5 minutes. May be the update interval is to short now? I'm going to try 6 minutes instead.

It may take me some time to sort these things out. The device has to run stable before investigating.

I'm going to a attach an external antenna and to exchange the SD-card. The power supply and cable should be fine, but I will exchange them too.

**Comment by DonKracho** [comment]
Now I got my device running stable. and investigated into the rate reading.

It looks to me it is the changeabsolut reding devided by the time intervall. That will give m³/min. But it was intended to be m³/h. In this case the value has to be normalized to one hour by multiplying it by 60.  rate[m³/h] =  changeabsolut * 60 / time interval.

Feature request: Please add the changeabsolute value to the mqtt json string too.

B.t.w. I designed just another ESP32-CAM Adapter for my watermeter: https://www.thingiverse.com/thing:5595270

**Reply by caco3** [reply]
> Feature request: Please add the changeabsolute value to the mqtt json string too.

Can you try if https://github.com/jomjol/AI-on-the-edge-device/actions?query=branch%3Aadd-changeabsolut-to-mqtt-json works for you?

**Reply by DonKracho** [reply]
Hi, thanks a lot. I did not try it yet, but probably not. If I have a look at the change you messed up the trailing brace of the json string. This can be fixed by moving the timestamp entry to the end again.

**Reply by caco3** [reply]
oh, right, sry, was to late yesterday 😒 
I fixed it now.

**Reply by caco3** [reply]
Note: In https://github.com/jomjol/AI-on-the-edge-device/pull/1246 MQTT ~change the rate from **per minute** to **per hour.**~ will add an additional topic which has various units for the value and rate.

**Comment by Apfelstruhdl** [comment]
Hi, I really enjoy the project, thanks for the great work!
I started using it for reading an old energy meter (Ferraris).

In that case the rate would make more sense to be normalized to watts.

Would it make sense to allow for an additional factor for the rate, so the rate can be adjusted to the application?

**Reply by caco3** [reply]
This better should be done in a post processing step, eg. in a Home Automation system.
